ERCS/erc-7555.md

```diff
@@ -0,0 +1,236 @@
+
+### Redirects
+An application looking to authenticate a user must navigate the user to a given provider's URI based on the `URI Request Syntax`. The application must implement a valid redirect URI for the callback in order to receive a valid response.
+
+#### Available Routes
+- `/auth/`: The route used to authneticate a user, and request credentials.
+- `/sendTransaction/`: The route used to send a transaction payload for a user to sign. This is more of a convenient method to allow applications to do both authentication, and plugin registration within a single redirect, instead of requiring the user to perform two redirects.
+
+### Schema
+The signer type represents the specific curve used when generating the keypair. The following are a non-exhaustive list of possible keypair types:
+```=
+type SIGNER_TYPE: string = "secp256k1" | "P256" | "BLS12-381";
+```
+The `signer_key` should be returned in the did:key format, as specified by the W3C.
+The `smart_account_address` should be returned in the CAIP-10 format.
+
+#### Auth Route
+##### Request Schema
+```=
+ parameters:
+    - in: query
+      name: redirect_uri
+      schema:
+        type: string
+      description: The uri that the provider should redirect back to.
+   - in: query
+      name: chain_id
+      schema:
+        type: string
+      description: The chain_id of a given network.
+```
+##### Response Schema
+```=
+ parameters:
+    - in: query
+      name: signer_key
+      schema:
+        type: string
+      description: The public key of the user, formatted using W3C definition of did:key
+  - in: query
+      name: smart_account_address
+      schema:
+        type: string
+      description: The on-chain address for a given smart account, formatted using CAIP-10
+```
+
+##### Request Syntax
+```=
+https://<PROVIDER_URI>/auth/?
+    redirect_uri=<YOUR_REDIRECT_URI>
+    &chain_id=<CHAIN_ID>
+```
+##### Response Syntax
+```=
+https://<YOUR_REDIRECT_URI>/auth/?
+    signer_key=<SIGNER_KEY>
+    &smart_account_address=<SMART_ACCOUNT_ADDRESS>
+```
+
+#### sendTransaction Route
+##### Request Schema
+```=
+ parameters:
+    - in: query
+      name: redirect_uri
+      schema:
+        type: string
+      description: The uri that the provider should redirect back to.
+   - in: query
+      name: chain_id
+      schema:
+        type: string
+      description: The chain_id of a given network.
+   - in: query
+      name: transaction
+      schema:
+        type: string
+      description: The RLP encoded transaction that needs to be signed
+```
+##### Response Schema
+```=
+ parameters:
+    - in: query
+      name: signer_key
+      schema:
+        type: string
+      description: The public key of the user, formatted using W3C definition of did:key
+  - in: query
+      name: smart_account_address
+      schema:
+        type: string
+      description: The on-chain address for a given smart account, formatted using CAIP-10
+  - in: query
+      name: tx_success
```

Comment:
Should the provider wait till tx is included in a block? Maybe it makes more sense to delegate this to the dapp (aka provider only returns tx_hash)

Reply:
I was thinking this would indicate that the TX was signed by the relayer? I'm not sure in that case we would even get the txhash from the provider? Might be making a large assumption.

Could we say the the userOp was signed and just go from there?

Reply:
Maybe we can go other way around, in case of successful signing, response is (txHash=0x12312312, error=undefined) in case of failure it's (txhash=undefined, error="something is wrong")?

Reply:
This seems fine to me, even if its not confirmed, having the txhash available for inclusion seems reasonable